packages/grant-explorer/src/features/api/passport.ts

```diff
@@ -1,65 +1,102 @@
+import { datadogLogs } from "@datadog/browser-logs";
 import {
-  submitPassport,
   fetchPassport,
   PassportResponse,
+  PassportResponseSchema,
   PassportState,
+  submitPassport,
 } from "common";
 import { useEffect, useState } from "react";
 
 export { submitPassport, fetchPassport, PassportState };
 export type { PassportResponse };
 
-export function usePassport({ address }: { address: string }) {
-  const [passport, setPassport] = useState<PassportResponse | undefined>();
+export function usePassport({ address }: { address: string | undefined }) {
   const [, setError] = useState<Response | undefined>();
 
   const [passportState, setPassportState] = useState<PassportState>(
     PassportState.LOADING
   );
 
+  const [passportScore, setPassportScore] = useState<number>();
+  const [passportColor, setPassportColor] = useState<
+    "orange" | "green" | "yellow"
+  >();
+  const [donationImpact, setDonationImpact] = useState<number>(0);
+
   useEffect(() => {
     setPassportState(PassportState.LOADING);
 
     const PASSPORT_COMMUNITY_ID =
       process.env.REACT_APP_PASSPORT_API_COMMUNITY_ID;
+
     if (PASSPORT_COMMUNITY_ID === undefined) {
       throw new Error("passport community id not set");
     }
-    const PASSPORT_THRESHOLD = 0;
 
     if (address && PASSPORT_COMMUNITY_ID) {
       const callFetchPassport = async () => {
         const res = await fetchPassport(address, PASSPORT_COMMUNITY_ID);
+
+        if (!res) {
+          datadogLogs.logger.error(
+            `error: callFetchPassport - fetch failed`,
+            res
+          );
+          setPassportState(PassportState.ERROR);
+          return;
+        }
+
         if (res.ok) {
-          const json = await res.json();
+          const scoreResponse = PassportResponseSchema.parse(await res.json());
 
-          if (json.status === "PROCESSING") {
+          if (scoreResponse.status === "PROCESSING") {
             console.log("processing, calling again in 3000 ms");
             setTimeout(async () => {
               await callFetchPassport();
             }, 3000);
             return;
-          } else if (json.status === "ERROR") {
-            // due to error at passport end
-            setPassportState(PassportState.ERROR);
+          }
+
+          if (
+            !scoreResponse.score ||
+            !scoreResponse.evidence ||
+            scoreResponse.status === "ERROR"
+          ) {
+            datadogLogs.logger.error(
+              `error: callFetchPassport - invalid score response`,
+              scoreResponse
+            );
+            setPassportState(PassportState.INVALID_RESPONSE);
             return;
           }
 
-          setPassport(json);
-          setPassportState(
-            json.score >= PASSPORT_THRESHOLD
-              ? PassportState.MATCH_ELIGIBLE
-              : PassportState.MATCH_INELIGIBLE
-          );
+          const score = Number(scoreResponse.evidence.rawScore);
```

Comment:
This could be done in the zod schema itself with `z.coerce.number()`

Reply:
Done.